lib/cartoRenderer.js

```diff
@@ -1430,7 +1430,8 @@ exports.processStyles = function(styles, assetsPath) {
           // Don't re-convert numbers or strings.
           if (!(typeof filter.val === 'number' || typeof filter.val === 'string')) {
             // If val doesn't have an explicit value, get one via toString.
-            filter.val = filter.val.value || filter.val.toString();
+            // Also, lets not make 0 a string
+            filter.val = filter.val.value === 0 ? filter.val.value : filter.val.value || filter.val.toString();
```

Comment:
Should we instead skip this block if `typeof filter.val.value === 'number'` ?
(and if so any undefined value checking needed there ?) I don't think so but going deep into untyped objects is always feels dangerous

Reply:
I don't believe so, I think what this is doing is making sure that `filter.val` is no longer an object but a primitive value

Reply:
sounds good to me